### PR TITLE
[ntuple] advertise RNTuple implements hadd MergeFile interface

### PR DESCRIPTION
**Pull request by mxxo**
Declares a new method `RNTuple::Merge` to conform to the `hadd` merger interface, namely: 
https://github.com/root-project/root/blob/331efa4c00fefc38980eaaf7b41b8e95fcd1a23b/io/doc/v530/index.html#L43-L53

I originally thought about to adding this method `RNTupleReader`, but think this is the right choice since it's the `RNTuple` blob that's stored on disk, and we have to parse it into a `RNTupleReader` to do any useful work. 

Actually running it segfaults, and I'm not sure whether I've made a mistake in linking or elsewhere. 
~~**Edit:** segfault fixed by #6016~~
**Edit**: I believe the segfault is fixed by 12fef499c109da4c2454d9cca2e60e88a883aa1c, where I avoid assuming that mergeable objects are derived from `TObject`.

### CODE REVIEW

File: tree/ntuple/v7/src/RMiniFile.cxx [hunk 1]
```diff
@@ -997,6 +998,14 @@ void ROOT::Experimental::Internal::RMiniFileReader::ReadBuffer(void *buffer, siz
 ////////////////////////////////////////////////////////////////////////////////
 
 
+Long64_t ROOT::Experimental::RNTuple::Merge(TCollection* /* input */, TFileMergeInfo* /* mergeInfo */) {
```
Comment (jblomer) [inline tree/ntuple/v7/src/RMiniFile.cxx:1001]: Let's define the method in a new compilation unit (e.g. `RNTupleMerge`).  This should make it easier to separate the mini RNTuple library from the full version.

File: io/io/src/TFileMerger.cxx [hunk 2]
```diff
@@ -573,26 +573,31 @@ Bool_t TFileMerger::MergeRecursive(TDirectory *target, TList *sourcelist, Int_t
...
+                              // Set ownership for collections
+                              if (hobj->InheritsFrom(TCollection::Class())) {
+                                 ((TCollection*)hobj)->SetOwner();
+                              }
+                              hobj->ResetBit(kMustCleanup);
+                              inputs.Add(hobj);
+                           } else {
+                             Info("MergeRecursive", "non-inheriting merge for key: %s in file %s",
+			        key->GetName(), nextsource->GetName());
+			   }
```
Comment (jblomer) [inline io/io/src/TFileMerger.cxx:593]: The whitespace seems odd

File: tree/ntuple/v7/src/RMiniFile.cxx (hunk 1 above)
Comment (mxxo) [reply tree/ntuple/v7/src/RMiniFile.cxx]: Fixed in fef9e17732a2a87a6d22b57e60d8db1cd9eaf43f

**Comment by mxxo** [comment]
I think this is ready for review. My intent is to fix the segfault and provide scaffolding for RNTuple merges in this PR and follow up with the actual implementation PRs. 

@pcanal are you comfortable with the error message printed on an attempted merge for non-inheriting objects? 

```text
hadd Target file: ../merged.root
hadd compression setting for all output: 1
hadd Source file 1: ntuple1.root
hadd Source file 2: ntuple2.root
hadd Sources and Target have different compression levels
hadd merging will be slower
hadd Target path: ../merged.root:/
Fatal in <TFileMerger::MergeRecursive>: Merging objects that don't inherit from TObject is unimplemented (key: ntuple in file ntuple2.root)
aborting
grep: warning: GREP_OPTIONS is deprecated; please use an alias or script
#0  0x00007f2dfd7ec07a in __GI___waitpid (pid=20605, stat_loc=stat_loc
entry=0x7fffc72e6c90, options=options
entry=0) at ../sysdeps/unix/sysv/linux/waitpid.c:29
#1  0x00007f2dfd764fbb in do_system (line=<optimized out>) at ../sysdeps/posix/system.c:148
#2  0x00007f2dfe92beb6 in TUnixSystem::StackTrace() () from /home/max/projects/root/new-errors/lib/libCore.so
#3  0x00007f2dfe7fe1aa in DefaultErrorHandler(int, bool, char const*, char const*) () from /home/max/projects/root/new-errors/lib/libCore.so
#4  0x00007f2dfe8b9be8 in ErrorHandler () from /home/max/projects/root/new-errors/lib/libCore.so
#5  0x00007f2dfe810e29 in TObject::Fatal(char const*, char const*, ...) const () from /home/max/projects/root/new-errors/lib/libCore.so
#6  0x00007f2dfeed49e3 in TFileMerger::MergeRecursive(TDirectory*, TList*, int) () from /home/max/projects/root/new-errors/lib/libRIO.so
#7  0x00007f2dfeed2e41 in TFileMerger::PartialMerge(int) () from /home/max/projects/root/new-errors/lib/libRIO.so
#8  0x0000000000406c8d in main::{lambda(TFileMerger&, int, int)#2}::operator()(TFileMerger&, int, int) const [clone .constprop.0] ()
#9  0x0000000000404ccd in main ()
#10 0x00007f2dfd740830 in __libc_start_main (main=0x403fa0 <main>, argc=5, argv=0x7fffc72ea358, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffc72ea348) at ../csu/libc-start.c:291
#11 0x0000000000406199 in _start ()
zsh: abort (core dumped)  ../bin/hadd -f ../merged.root ntuple1.root ntuple2.root
```

**Comment by mxxo** [comment]
Blocked on #6073 because of the case handling logic there.

**Comment by mxxo** [comment]
@pcanal I reworked the error message for unimplemented merges in 0ccd1d3, thank you for the feedback :)

**Pull request merged**